File: protobuf_mqtt_bridge/examples/python/mqtt_talker.py
import paho.mqtt.client
import math
import time
import autoware_protobuf.geometry_msgs.msg.pose_stamped_pb2

def create_mqtt_client():
    client = paho.mqtt.client.Client()
    client.on_connect = on_connect
    client.connect("localhost", 1883, 60)
    return client

def on_connect(client, userdata, flags, result):
    print(f"Connected: {result}")

def create_message(count):
    stamp = math.modf(time.time())
    msg = autoware_protobuf.geometry_msgs.msg.pose_stamped_pb2.PoseStamped()
    msg.header.stamp.sec = int(stamp[1])
    msg.header.stamp.nanosec = int(stamp[0] * 1e9)
    msg.header.frame_id = "map"
    msg.pose.position.x = math.cos(math.radians(count))
    msg.pose.position.y = math.sin(math.radians(count))
    msg.pose.orientation.z = math.sin(math.radians(count/2.0))
    msg.pose.orientation.w = math.cos(math.radians(count/2.0))
    return msg

def publish_loop(client):
    try:
        count = 0
        while True:
            message = create_message(count)
            print(message)
            client.publish("/pose_stamped", message.SerializeToString())
            count = (count + 10) % 360
            time.sleep(0.1)
    except KeyboardInterrupt:
        pass

if __name__ == "__main__":
    client = create_mqtt_client()
    client.loop_start()
    publish_loop(client)
    client.loop_stop()

File: protobuf_mqtt_bridge/examples/node/mqtt_talker.js
const autoware_protobuf = require('./autoware_protobuf');
const mqtt = require('mqtt')
const client  = mqtt.connect('mqtt://localhost')

client.on('connect', function () {
    console.log('connected');
})

let count = 0
const loop = function()
{
    const PoseStamped = autoware_protobuf.geometry_msgs.msg.PoseStamped;
    data = {
        "header": {"stamp": {"sec": 1, "nanosec": 2}, "frame_id": "map"},
        "pose": {
            "position": {
                "x": Math.cos(count * Math.PI / 180),
                "y": Math.sin(count * Math.PI / 180),
            },
            "orientation": {
                "z": Math.sin((count + 180) / 2.0 * Math.PI / 180),
                "w": Math.cos((count + 180) / 2.0 * Math.PI / 180),
            },
        }
    }
    const err = PoseStamped.verify(data);
    if (err) throw Error(err);

    const payload = PoseStamped.encode(data).finish();
    client.publish("/pose_stamped", payload);
    count = (count + 10) % 360
}

setInterval(loop, 100);
